dashboard/app/models/featured_project.rb

```diff
@@ -5,7 +5,9 @@
 #  id                   :integer          not null, primary key
 #  storage_app_id       :integer
 #  who_featured_user_id :integer
-#  created_at           :datetime
+#  featured_at          :datetime
+#  unfeatured_at        :datetime
+#  is_featured          :boolean
```

Comment:
Do we need all 3 of these? Can you explain how it will be used? At the very least, `is_featured` can be deduced (and better provided in a more Rails-y `featured?` model method) by seeing which of the other 2 is more recent. 

Are we interested in keeping both the latest featured and unfeatured dates? Presumably we don't care about any further history, including who unfeatured or who featured previously if multiple times.

Reply:
There's going to be an archive list of previously featured projects:
<img width="827" alt="screen shot 2018-01-29 at 2 50 08 pm" src="https://user-images.githubusercontent.com/12300669/35538720-b9267edc-0503-11e8-8ced-cc582a05c779.png">

I need to track when the project was most recently featured and when (if ever) it was most recently unfeatured. History beyond the most recent time of (un)featuring is unnecessary.  I also don't care about who unfeatured the project, and I'm not really using information about who most recently featured the project either; possibly a reason to delete that field?

Reply:
Ok, so we want the 2 dates. Can we remove the bool column and deduce its value?

Reply:
if we do not need to know when a featured (then un-featured, then re-featured) project was last unfeatured, then it seems like we could potentially get rid of the `is_featured` column and have the database query for featured projects look for `featured_at IS NOT NULL and unfeatured_at IS NULL`. This would be my recommendation, since it eliminates duplicate state.

Andrew, please correct me if I'm wrong, but I would be worried that a query which did a date comparison across two indexed columns would be slower than a query which simply does a null (or not null) check on two indexed columns.

as for the user column, this one is really up to you. I'd originally recommended that you keep this column so that you can track down who featured a project, but the [YAGNI](https://martinfowler.com/bliki/Yagni.html) principle says never to add this kind of thing until you know for sure you will need it.

Reply:
To be clear, we need to be able to query the database efficiently for a list of all featured (&& not yet unfeatured) projects in order to display them on the /projects/public page.

Reply:
I included the is_featured column because I wanted to easily be able to pull all of the featured projects and all of the previously (but not currently featured) projects, which seemed quicker and more straightforward with a separate column than with a date comparison for each featured project. I can change to a more conventional `featured?` method, though.

Reply:
To further clarify, I'm suggesting that we null out the `unfeatured_at` column when re-featuring a previously un-featured project.

Reply:
Good call. I wasn't thinking about querying. I agree @davidsbailey, we should query as you suggested with `IS NULL` rather than comparing specific dates.

Then, if something is re-featured after bing unfeatured, we will need to nullify `unfeatured_at`.

Reply:
@Erin007 The `featured?` method won't help you with queries, just for that value on a specific instance of the model.

Dave's query will get you the currently featured list, and you can do a similar query where each `_at` field `IS NOT NULL` to get the previously (but not currently) featured list.

Reply:
Note you can create [scopes](http://guides.rubyonrails.org/active_record_querying.html#scopes) for these common queries in the model.